cmd/oidc-token-verifier/main.go

```diff
@@ -0,0 +1,202 @@
+
+	// Print used options values.
+	logger.Infow("Using the following trusted workflows", "trusted-workflows", opts.trustedWorkflows)
+	logger.Infow("Using the following client ID", "client-id", opts.clientID)
+	logger.Infow("Using the following public key path", "public-key-path", opts.publicKeyPath)
+	logger.Infow("Using the following new public keys environment variable", "new-keys-var", opts.newPublicKeysVarName)
+	logger.Infow("Using the following claims output path", "claims-output-path", opts.outputPath)
+
+	verifyConfig, err := tioidc.NewVerifierConfig(logger, opts.clientID)
+	if err != nil {
+		return err
+	}
+	logger.Infow("Verifier config created", "config", verifyConfig)
+
+	// TODO(dekiel): add support for providing trusted issuers instead of using the value from the package.
+	tokenProcessor, err := tioidc.NewTokenProcessor(logger, tioidc.TrustedOIDCIssuers, opts.token, verifyConfig)
+	if err != nil {
+		return err
+	}
+	logger.Infow("Token processor created for trusted issuer", "issuer", tokenProcessor.Issuer())
+
+	ctx := context.Background()
+	provider, err := tioidc.NewProviderFromDiscovery(ctx, logger, tokenProcessor.Issuer())
+	if err != nil {
+		return err
+	}
+	logger.Infow("Provider created using OIDC discovery", "issuer", tokenProcessor.Issuer())
+
+	verifier := provider.NewVerifier(logger, verifyConfig)
+	logger.Infow("New verifier created")
+
+	claims := tioidc.Claims{}
+	err = tokenProcessor.Claims(ctx, &verifier, claims)
+	if err != nil {
+		return err
+	}
+	logger.Infow("Token verified, extracted claims", "claims", claims)
+	// TODO: verify claims values against expected values
+
+	// write claims to file
+	file, _ := os.Create(opts.outputPath)
+	defer file.Close()
+	encoder := json.NewEncoder(file)
+	err = encoder.Encode(&claims)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// If the public keys are not cached or expired, it uses OIDC discovery to get the public keys.
+// New public keys are written to the file specified by the --public-key-path flag.
+// If new public keys are fetched, it sets ado environment variable to true.
+
+// loadPublicKeysFromLocal loads the public keys from the file specified by the --public-key-path flag.
+// example implementation https://gist.github.com/nilsmagnus/199d56ce849b83bdd7df165b25cb2f56
+// func (opts *options) loadPublicKeysFromLocal() error {
+//
+// }
+//
+
+// savePublicKeysFromRemote fetches the public keys from the OIDC discovery endpoint.
+// It writes the public keys to the file specified by the --public-key-path flag.
+// It sets the environment variable specified by --new-public-keys-var-name to true to indicate that new public keys are fetched.
+// func (opts *options) savePublicKeysFromRemote(issuer string) error {
+//
+// }
+
+// setAdoEnvVar sets the Azure DevOps pipeline environment variable to true.
+// Environment variable name is specified by --new-public-keys-var-name flag.
+// func (opts *options) setAdoEnvVar() error {
+//
+// }
```

Comment:
should these comments be in this place? They look like leftovers for commented functions.

Reply:
Yes, they should be in a file. These are comment for feature of using cached public keys.